src/btree/node.ts

```diff
@@ -232,6 +237,9 @@ export class DataNodeImpl extends NodeImpl<ReadonlyJSONValue> {
       limit > 0 && i < entries.length && entries[i][0].startsWith(prefix);
       limit--, i++
     ) {
+      if (onKey) {
+        onKey(entries[i][0], limit === 1, i === entries.length - 1);
```

Comment:
I do not understand why we have the third argument (`isLastEntry`) here? It only means that it is the last entry in the current chunk/DataNode. There might be more entries in the next chunk/DataNode.

Given a tree like this:

```
              InternalNode
      [['b', hash1], ['d', hash2]]
         /                    \
     DataNode              DataNode
[['a', 0, ['b', 1]]   [['c', 2, ['d', 3]]
````

The key `'b'` has no significance in `scan({})`

Reply:
I see.  

I've removed this as it was only needed for the version that tried to optimize the case where the scan was not read to limit.  

In particular, if the scan ran out of entries before hitting limit, then we can't use the lastKey to rule out matches (because items can be added after the very last entry that should match the scan and our greater than lastKey, its very similar to why we agreed you can only use lastKey for prefix if the limit was hit).  

How would we determine if it is the very last entry (not in this chunk, but overall)?